src/main/test/cz/muni/fi/DaoTests/DriverDaoTest.java

```diff
@@ -0,0 +1,106 @@
+import cz.fi.muni.pa165.PersistenceSampleApplicationContext;
+import java.util.List;
+import javax.persistence.EntityManager;
+import javax.persistence.EntityManagerFactory;
+import javax.persistence.PersistenceContext;
+import javax.persistence.PersistenceUnit;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.testng.AbstractTestNGSpringContextTests;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+import static org.assertj.core.api.Assertions.*;
+
+
+/**
+ * @author Lucie Kureckova, 445264
+ */
+@ContextConfiguration(classes = PersistenceSampleApplicationContext.class)
+public class DriverDaoTest extends AbstractTestNGSpringContextTests {
```

Comment:
There's problem in your tests,  you can't count on executing tests in order as they are in your class! What can happen is that `updateDriverTest()` will be executed as first, that would lead to failing `listTestDriversTest()` bcs in DB would be already driver2 (with default isMainDriver = false) but you check only for testDriver1 and testDriver2, and so on with different tests.